lib/tunnel.js

```diff
@@ -936,5 +959,84 @@ function sleep(sleepDuration) {
     /* do nothing */
   }
 }
+/**
+ * getTunnelID is used to Run Tunnel Binary
+ * @param {number} infoAPIPort running local Server Port.
+ * @param {number} retries max attempt, retries to get port.
+ * @return {number/boolean} Return id/false whether tunnel is Running Successfully or
+ *     not.
+ */
+async function getTunnelID(infoAPIPort, retries){
+  try {
+    if (retries > 0) {
+      var url = 'http://127.0.0.1:' + infoAPIPort + '/api/v1.0/info';
+      
+      try {
+        const response = await axios.get(url)
+        
+        if (response.data && response.data.data && response.data.data.id){
+          let tunnelID = response.data.data.id
+          console.log("tunnel started with ID: ", tunnelID)
+          return tunnelID
+
+        }
+        
+      } catch (_) {
+      }
+      await new Promise(r => setTimeout(r, 2000));
+      console.log("check tunnel status attempts left: ", retries - 1);
+      getTunnelID(infoAPIPort, retries - 1)
```

Comment:
add return statement to this recursive call else the value wont be sent back from callstack.

Reply:
done